Linker errors for static build due to a missing C++ stdlib

I encountered the following linker errors when trying to build a static version of SPRAL in a [Conan recipe](https://github.com/valgur/conan-center-index/blob/dev/recipes/spral/all/conanfile.py):
```
... 500 more lines ...
/usr/bin/ld: libspral.a(src_ssids_cpu_kernels_wrappers.cxx.o): in function `int spral::ssids::cpu::lapack_sytrf<double>(spral::ssids::cpu::fillmode, int, double*, int, int*, double*, int) [clone .cold]':
wrappers.cxx:(.text.unlikely+0x79): undefined reference to `__cxa_allocate_exception'
/usr/bin/ld: wrappers.cxx:(.text.unlikely+0x8b): undefined reference to `std::runtime_error::runtime_error(char const*)'
/usr/bin/ld: wrappers.cxx:(.text.unlikely+0xa2): undefined reference to `std::runtime_error::~runtime_error()'
/usr/bin/ld: wrappers.cxx:(.text.unlikely+0xa9): undefined reference to `typeinfo for std::runtime_error'
/usr/bin/ld: wrappers.cxx:(.text.unlikely+0xb1): undefined reference to `__cxa_throw'
/usr/bin/ld: wrappers.cxx:(.text.unlikely+0xc0): undefined reference to `__cxa_free_exception'
collect2: error: ld returned 1 exit status
```

This appears to be caused by `gfortran` getting incorrectly picked as the linker because of `link_language : 'fortran'` being set for the library and executable targets in `meson.build`. Removing these and letting Meson pick the correct linker automatically fixed it for me.

----
Here's my `conan_meson_native.ini` for completeness:
```ini
[properties]

[constants]
preprocessor_definitions = []

[project options]
wrap_mode = 'nofallback'
bindir = 'bin'
sbindir = 'bin'
libexecdir = 'bin'
includedir = 'include'
libdir = 'lib'
tests = false
examples = false
modules = false
openmp = True
gpu = False
libmetis_version = '5'


[binaries]
fc = 'gfortran-13'
c = 'gcc-13'
cpp = 'g++-13'

[built-in options]
buildtype = 'release'
default_library = 'static'
b_ndebug = 'true'
cpp_std = 'c++17'
backend = 'ninja'
pkg_config_path = '/home/martin/.conan2/p/b/spral5a7b73ea217ed/b/build-release/conan'
# C/C++ arguments
c_args = ['-m64'] + preprocessor_definitions
c_link_args = ['-m64']
cpp_args = ['-m64'] + preprocessor_definitions
cpp_link_args = ['-m64']
```

--- a/meson.build
+++ b/meson.build
@@ -144,6 +144,9 @@ if build_openmp
   add_global_link_arguments(lomp, language : 'fortran')
   add_global_link_arguments(lomp, language : 'c')
   add_global_link_arguments(lomp, language : 'cpp')
+  if nvcc_available
+    add_global_link_arguments(lomp, language : 'cuda')
+  endif
 endif
 
 binspral_src = []

--- a/meson.build
+++ b/meson.build
@@ -146,20 +146,6 @@ if build_openmp
   add_global_link_arguments(lomp, language : 'cpp')
 endif
 
-# Link flag for C++
-lstdcpp = '-lstdc++'
-if host_machine.system() != 'linux'
-  if cxx.get_id() == 'clang' or cxx.get_id() == 'clang-cl'
-    lstdcpp = '-lc++'
-  endif
-  if cxx.get_id() == 'intel' or cxx.get_id() == 'intel-llvm'
-    lstdcpp = '-cxxlib'
-  endif
-  if cxx.get_id() == 'intel-cl' or cxx.get_id() == 'intel-llvm-cl'
-    lstdcpp = '/Qcxxlib'
-  endif
-endif
-
 binspral_src = []
 libspral_src = []
 libspral_c_src = []
@@ -192,8 +178,6 @@ subdir('tests')
 libspral = library('spral',
                    sources : libspral_src + libspral_c_src + libspral_cpp_src + libspral_nvcc_src,
                    dependencies : libspral_deps,
-                   link_language : 'fortran',
-                   link_args : lstdcpp,
                    include_directories: libspral_include,
                    install : true)
 
@@ -202,7 +186,6 @@ executable('spral_ssids',
            sources : binspral_src,
            dependencies : libcuda,
            link_with : libspral,
-           link_language : 'fortran',
            install : true)
 
 # Headers
@@ -225,8 +208,10 @@ if build_tests
     name = test[1]
     file = test[2]
     test(name,
-         executable(name, file, link_with : libspral, dependencies : libspral_deps,
-                    link_language : 'fortran', include_directories: libspral_include),
+         executable(name, file,
+                    link_with : libspral,
+                    dependencies : libspral_deps,
+                    include_directories: libspral_include),
          suite : [package, 'tests', 'fortran'],
          timeout : 300, is_parallel : false, env: omp_env)
   endforeach
@@ -236,8 +221,10 @@ if build_tests
     name = test[1]
     file = test[2]
     test(name,
-         executable(name, file, link_with : libspral, dependencies : libspral_deps,
-                    link_language : 'c', include_directories : libspral_include),
+         executable(name, file,
+                    link_with : libspral,
+                    dependencies : libspral_deps,
+                    include_directories : libspral_include),
          suite : [package, 'tests', 'C'],
          timeout : 300, is_parallel : false, env: omp_env)
   endforeach
@@ -247,8 +234,10 @@ if build_tests
     name = test[1]
     file = test[2]
     test(name,
-         executable(name, file, link_with : libspral, dependencies : libspral_deps,
-                    link_language : 'cpp', include_directories : libspral_include),
+         executable(name, file,
+                    link_with : libspral,
+                    dependencies : libspral_deps,
+                    include_directories : libspral_include),
          suite : [package, 'tests', 'C++'],
          timeout : 300, is_parallel : false, env: omp_env)
   endforeach
@@ -264,9 +253,12 @@ if build_examples
     name = example[1]
     file = example[2]
     test(name,
-         executable(name, file, link_with : libspral, dependencies : libspral_deps,
-                    link_language : 'fortran', include_directories : libspral_include,
-                    install : true, install_dir : fortran_examples_folder),
+         executable(name, file,
+                    link_with : libspral,
+                    dependencies : libspral_deps,
+                    include_directories : libspral_include,
+                    install : true,
+                    install_dir : fortran_examples_folder),
          suite : [package, 'examples', 'fortran'],
          timeout : 300, is_parallel : false, env: omp_env)
   endforeach
@@ -278,9 +270,12 @@ if build_examples
     name = example[1]
     file = example[2]
     test(name,
-         executable(name, file, link_with : libspral, dependencies : libspral_deps,
-                    link_language : 'c', include_directories : libspral_include,
-                    install : true, install_dir : c_examples_folder),
+         executable(name, file,
+                    link_with : libspral,
+                    dependencies : libspral_deps,
+                    include_directories : libspral_include,
+                    install : true,
+                    install_dir : c_examples_folder),
          suite : [package, 'examples', 'C'],
          timeout : 300, is_parallel : false, env: omp_env)
   endforeach

